cpp/src/parquet/arrow/record_reader.cc

```diff
@@ -448,36 +448,17 @@ class RecordReader::RecordReaderImpl {
   std::shared_ptr<::arrow::ResizableBuffer> rep_levels_;
 };
 
-template <typename DType>
-struct RecordReaderTraits {
-  using BuilderType = ::arrow::ArrayBuilder;
-};
-
-template <>
-struct RecordReaderTraits<ByteArrayType> {
-  using BuilderType = ::arrow::internal::ChunkedBinaryBuilder;
-};
-
-template <>
-struct RecordReaderTraits<FLBAType> {
-  using BuilderType = ::arrow::FixedSizeBinaryBuilder;
-};
-
 template <typename DType>
 class TypedRecordReader : public RecordReader::RecordReaderImpl {
  public:
   using T = typename DType::c_type;
 
-  using BuilderType = typename RecordReaderTraits<DType>::BuilderType;
-
   TypedRecordReader(const ColumnDescriptor* descr, ::arrow::MemoryPool* pool)
-      : RecordReader::RecordReaderImpl(descr, pool), current_decoder_(nullptr) {
-    InitializeBuilder();
-  }
+      : RecordReader::RecordReaderImpl(descr, pool), current_decoder_(nullptr) {}
 
   void ResetDecoders() override { decoders_.clear(); }
 
-  inline void ReadValuesSpaced(int64_t values_with_nulls, int64_t null_count) {
+  virtual inline void ReadValuesSpaced(int64_t values_with_nulls, int64_t null_count) {
```

Comment:
If this is virtual then `inline` probably should not be here

Reply:
Thanks I wasn't too sure about that.  I found this [reference](http://www.cs.technion.ac.il/users/yechiel/c++-faq/inline-virtuals.html) which didn't help resolve my indecision.